src/mlpack/methods/ann/layer/radial_basis_function_impl.hpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file radial_basis_impl.hpp
+ * @author Himanshu Pathak
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "dropout.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+RBF<InputDataType, OutputDataType>::RBF() :
+    inSize(0),
+    outSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+RBF<InputDataType, OutputDataType>::RBF(
+    const size_t inSize,
+    const size_t outSize) :
+    inSize(inSize),
+    outSize(outSize)
+{
+  centres = arma::mat(inSize, outSize, arma::fill::randu);
+}
+
+template<typename InputDataType, typename OutputDataType>
+void RBF<InputDataType, OutputDataType>::Reset()
+{
+  centres = arma::normcdf(centres, 0, 1);
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void RBF<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>& input,
+    arma::Mat<eT>& output)
+{
+  arma::cube x = arma::cube(inSize, outSize, input.n_cols);
+
+  for (size_t i = 0; i < input.n_cols; i++)
+  {
+    x.slice(i).each_col() = input.col(i);
+  }
```

Comment:
I think we can implement this without creating the cube `x` as I previously suggested. It consumes extra memory. Let me know what you think.